data/transactions/logic/eval.go

```diff
@@ -626,6 +626,18 @@ func opErr(cx *evalContext) {
 	cx.err = errors.New("TEAL runtime encountered err opcode")
 }
 
+func opReturn(cx *evalContext) {
+	// Achieve the end condition:
+	// Take the last element on the stack and make it the return value (only element on the stack)
+	// Move the pc to the end of the program
+	last := len(cx.stack) - 1
+	if last != 0 {
```

Comment:
I don't think you need this `if`, right?

Reply:
simplified